servers/rendering/rendering_device.cpp

```diff
@@ -404,7 +412,24 @@ void RenderingDevice::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("uniform_set_create", "uniforms", "shader", "shader_set"), &RenderingDevice::_uniform_set_create);
 	ClassDB::bind_method(D_METHOD("uniform_set_is_valid", "uniform_set"), &RenderingDevice::uniform_set_is_valid);
 
-	ClassDB::bind_method(D_METHOD("buffer_update", "buffer", "offset", "size_bytes", "data", "post_barrier"), &RenderingDevice::_buffer_update, DEFVAL(BARRIER_MASK_ALL));
+	ClassDB::bind_method(D_METHOD("buffer_update", "buffer", "offset", "size", "data", "post_barrier"), &RenderingDevice::_buffer_update_array<uint8_t>, DEFVAL(BARRIER_MASK_ALL));
+	ClassDB::bind_method(D_METHOD("buffer_update_int32", "buffer", "offset", "size", "data", "post_barrier"), &RenderingDevice::_buffer_update_array<int32_t>, DEFVAL(BARRIER_MASK_ALL));
+	ClassDB::bind_method(D_METHOD("buffer_update_int64", "buffer", "offset", "size", "data", "post_barrier"), &RenderingDevice::_buffer_update_array<int64_t>, DEFVAL(BARRIER_MASK_ALL));
+	ClassDB::bind_method(D_METHOD("buffer_update_float32", "buffer", "offset", "size", "data", "post_barrier"), &RenderingDevice::_buffer_update_array<float>, DEFVAL(BARRIER_MASK_ALL));
+	ClassDB::bind_method(D_METHOD("buffer_update_float64", "buffer", "offset", "size", "data", "post_barrier"), &RenderingDevice::_buffer_update_array<double>, DEFVAL(BARRIER_MASK_ALL));
+	ClassDB::bind_method(D_METHOD("buffer_update_vector2", "buffer", "offset", "size", "data", "post_barrier"), &RenderingDevice::_buffer_update_array<Vector2>, DEFVAL(BARRIER_MASK_ALL));
+	ClassDB::bind_method(D_METHOD("buffer_update_vector3", "buffer", "offset", "size", "data", "post_barrier"), &RenderingDevice::_buffer_update_array<Vector3>, DEFVAL(BARRIER_MASK_ALL));
+	ClassDB::bind_method(D_METHOD("buffer_update_color", "buffer", "offset", "size", "data", "post_barrier"), &RenderingDevice::_buffer_update_array<Color>, DEFVAL(BARRIER_MASK_ALL));
```

Comment:
I would remove the Vector3 version because alignment is not possible, so it will never be used.